modules/ROOT/pages/http/http-listener-ref.adoc

```diff
@@ -1,182 +1,296 @@
 
-Put a wildcard at the end of a path to prevent unexpected results when specific paths configured in other flows override the wildcard path. For example, the Path settings for the HTTP Listener in flows are set as follows:
+Take the following example paths for three listeners using a configuration that establishes `api/v1` as base path:
 
-* Path configured in Flow A: `/regions/*/customers`
-* Path configured in Flow B: `/regions/us/prospects`
+* `account/mulesoft/main-contact`: will only match the exact path request `http://awesome-company.com/api/v1/account/mulesoft/main-contact`
+* `account/{accountId}/main-contact`: will match all path requests structured similarly such as `http://awesome-company.com/api/v1/account/salesforce/main-contact` and save `salesforce` as the value of `accountId`.
+* `account/{accountId}/*`: will match all path requests different from `main-contact` such as `http://awesome-company.com/api/v1/account/mulesoft/users` and save `mulesoft` as the value of `accountId`.
 
-A request to `/regions/us/customers` fails and is routed to flow B.
+[TIP]
+Wildcards at the end of a path can help capture requests to unhandled resources to provide better error messages.
 
-Put a wildcard in the middle of a path when you want a request to succeed that has a URL consisting of the prefix and suffix to the asterisk. For example, Path = `account/\*/main-contact`. The HTTP Listener listens for every HTTP request having a URI that starts with `/account/` and ends with `/main-contact`; the segment in the middle can vary. Requests succeed from the following URLs:
+When multiple listeners are defined, requests are always routed to the most specific one. Hence in the previous examples a request with `accountId` `mulesoft` and suffix `main-contact` will be handled by the first listener,
+and any different `accountId` value by the second listener instead of the third.
 
-* `+http://localhost:8081/account/mulesoft/main-contact+`
-* `+http://localhost:8081/account/foo/main-contact+`
+=== About Methods
 
-=== Using a Placeholder in a URI
+Request can also be routed based on the HTTP method received. By default, a listener will handle all methods but these can be restricted. In the following example, `GET`, `POST` and `PUT` requests will be routed to different flows.
+This is useful to restrict the visibility of data: we may allow certain users to see our data but only a few to modify it.
 
-You can define a path that uses a placeholder in curly braces: `{accountId}` for example, `account/{accountId}/main-contact`. The connector maps the accountId to a URI parameter attribute in the flow.
+[source,xml,linenums]
+----
+<flow name="main-contact-write">
+  <http:listener path="account/{accountId}/main-contact" allowedMethods="POST, PUT" config-ref="HTTP_Listener_config"/>
+  <!-- validate user permissions -->
```

Comment:
I don't agree, even if I add an excellent example no one will remember to update the related components used here. The examples will be destined to become obsolete and then we'll have users asking questions about the example they copied and are not working. 
If this was a mule cookbook like the one DW has it would make sense, but this is HTTP exclusive.